### PR TITLE
Refactor header to use native blocks and add some translucency

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

This PR aims to refactor the model header of our theme. It uses the new sticky-layout and replaces the specific `header-row` and `header-spacer` blocks with native ones, such as `flex-layout`.

It also adds a gentle animation when the header sticks to the top of the page and makes it a little bit translucent.

#### How should this be manually tested?

Compare my [workspace](https://kiwi--storecomponents.myvtex.com/) with the [original one](http://storetheme.vtex.com/)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
✔️ | Technical improvements

### CODE REVIEW

File: store/blocks/header/header.jsonc [hunk 1]
```diff
@@ -7,21 +7,21 @@
...
   "header-layout.desktop": {
     "children": [
-      "header-row#1-desktop",
-      "header-row#2-desktop",
-      "header-row#3-desktop",
-      "header-row#4-desktop"
+      "flex-layout.row#1-desktop",
+      "flex-layout.row#2-desktop",
+      "flex-layout.row#3-desktop",
+      "sticky-layout.stack-container#4-desktop"
```
Comment (klzns) [inline store/blocks/header/header.jsonc]: Why is that needed? I think it could be if you use more than one sticky-layout right? But in this case it's not needed, right?

Comment (kaisermann) [reply store/blocks/header/header.jsonc]: Right, it's not needed, but I tried keeping the same functionalities the previous header had. Do you prefer if I remove it?

File: styles/css/vtex.store-header.css [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+  transition: all 0.3s ease;
+  display: block;
 }
 
-.headerRow--main-header .headerRowBackground {
-  background-color: #F0F0F0;
+/* desktop/mobile main header padding and background transitions */
+:global(.vtex-flex-layout-0-x-flexRowContent--main-header),
+:global(.vtex-flex-layout-0-x-flexRowContent--main-header-mobile) {
+  transition: background 0.3s ease, box-shadow 0.3s ease;
```
Comment (lbebber) [inline styles/css/vtex.store-header.css]: por mim eu desabilitava essas transiçoes, pq sao transiçoes nao performaticas e nao queria incentivar as pessoas a fazerem isso. mas fica a teu criterio

Comment (kaisermann) [reply styles/css/vtex.store-header.css]: era just for show, posso tirar sim
